tests/nunit_tools/nunit_meta_runner/Program.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Collections.Generic;
+using System.Diagnostics;
+using NUnitLite;
+using NUnit.Framework.Api;
+using NUnit.Framework.Interfaces;
+
+// preloading dependencies
+using Trinity;
+using NUnit.Framework;
+
+namespace NUnitMetaRunner
+{
+    class Program
+    {
+        public static int Main(string[] args)
+        {
+            if (args.Length < 2)
+            {
+                Console.Error.WriteLine("Usage: runner <runner-path> <test-assembly> <result-dir> [nunitlite-arg ..]");
+                return 1;
+            }
+
+            var runnerPath = Path.GetFullPath(args[0]);
+            var resultDirPath = Path.GetFullPath(args[1]);
+            var assemblyPath = Path.GetFullPath(args[2]);
+            var remainingOptions = args.Skip(3).ToArray();
+
+            // NOTE(leasunhy): this may fail silently if some of the dependencies are not preloaded!
```

Comment:
Yes, in our scenario, when NUnit is not preloaded, it will throw an exception, but when Trinity is not preloaded, it succeeds but no tests can be discovered.